Geometry/HGCalCommonData/plugins/dd4hep/DDHGCalHEAlgo.cc

```diff
@@ -0,0 +1,591 @@
+///////////////////////////////////////////////////////////////////////////////
+// File: DDHGCalHEAlgo.cc
+// Description: Geometry factory class for HGCal (Mix)
+// Author : Raman Sehgal
+// DD4hep code for, HGCalHEAlgo, developed by Sunanda Banerjee
+///////////////////////////////////////////////////////////////////////////////
+
+#include <cmath>
+#include <memory>
+#include <string>
+#include <unordered_set>
+#include <vector>
+
+#include "Geometry/HGCalCommonData/interface/HGCalParameters.h"
+#include "Geometry/HGCalCommonData/interface/HGCalGeomTools.h"
+#include "Geometry/HGCalCommonData/interface/HGCalWaferType.h"
+#include "DD4hep/DetFactoryHelper.h"
+#include "DataFormats/Math/interface/CMSUnits.h"
+#include "DetectorDescription/DDCMS/interface/DDPlugins.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+
+//#define EDM_ML_DEBUG
+using namespace cms_units::operators;
+
+struct HGCalHEAlgo {
+  HGCalHEAlgo() { throw cms::Exception("HGCalGeom") << "Wrong initialization to HGCalHEAlgo"; }
```

Comment:
@bsunanda a similar argument as for other PRs with this constructor declaration applies